### DISCUSSION TITLE
Resize image to "Fit Contain" the given width/height

### DISCUSSION THREAD

**Discussion by andrefelipe** [libvips/php-vips#182] [Q&A] [answered]
So embarrassed to ask this, I did look a lot myself, in web, in Sharp source code and libvips too, sorry.

How to do in PHP a resize of a image to contain inside an area and fill the empty space with a color?

The result I need is:
— Resize a image to contain inside a 1080 x 1080px area (centered);
— Fill the background space with #222222 color;
— The resulting image is always 1080 x 1080px;

Exactly like the Sharp resize API, with the "fit" option:
https://sharp.pixelplumbing.com/api-resize#resize

**Comment by jcupitt** [comment, marked as answer]
Hey @andrefelipe,

Use `thumbnail` to fit within 1080x1080 (so one axis is cropped), then use `gravity` to expand to exactly the size you need. Perhaps (untested):

```php
$image = Vips\Image::thumbnail("some-image-file.xxx", 1080);
$image = $image->gravity("centre", 1080, 1080, ["extend" => "background", "background" => 86]);
$image->WriteToFile("thumb.jpg");
```

**Reply by andrefelipe** [reply]
Thank you very much!

What is the color format for the "background" option?

I tried HEX "#222222" (with or without #) and tried too "34, 34, 34" but didn't work.

**Reply by jcupitt** [reply]
Just use `"background" => 86`. 22h is 86 in decimal. You can also use `"background" => [86, 86, 86]`, but if the three numbers are the same you can use a simple scalar.

**Reply by andrefelipe** [reply]
OK, got it! 

It's RGB then. 

I confused because HEX 222222 is actually RGB 34, 34, 34.
(RGB 86, 86, 86 is a lot brighter)

I'm using `background" => 34` then.

Thank you very very much for your time and attention. Have a pleasant Sunday.

**Reply by jcupitt** [reply]
Ooop, brain fart, you're right, 34 is the correct value.
